src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -2694,6 +2642,233 @@ def get_enable_azure_rbac(self, enable_validation: bool = False, **kwargs) -> bo
+                    raise MutuallyExclusiveArgumentError(
+                        "--dns-name-prefix and --fqdn-subdomain cannot be used at same time"
+                    )
+                private_dns_zone = self.get_private_dns_zone()
+                if private_dns_zone:
+                    if private_dns_zone.lower() != CONST_PRIVATE_DNS_ZONE_SYSTEM:
+                        if not is_valid_resource_id(private_dns_zone):
+                            raise InvalidArgumentValueError(
+                                private_dns_zone + " is not a valid Azure resource ID."
+                            )
+                    raise InvalidArgumentValueError(
+                        "--fqdn-subdomain should only be used for private cluster with custom private dns zone"
+                    )
+        return fqdn_subdomain
+
+    def get_fqdn_subdomain(self) -> Union[str, None]:
+        """Obtain the value of fqdn_subdomain.
+
+        This function will verify the parameter by default. It will check if both dns_name_prefix and fqdn_subdomain
+        are assigend, if so, raise the MutuallyExclusiveArgumentError. It will also check when both private_dns_zone
+        and fqdn_subdomain are assigned, if the value of private_dns_zone is CONST_PRIVATE_DNS_ZONE_SYSTEM, raise an
+        InvalidArgumentValueError; Otherwise if the value of private_dns_zone is not a valid resource ID, raise an
+        InvalidArgumentValueError.
+
+        :return: string or None
+        """
+
+        return self._get_fqdn_subdomain(enable_validation=True)
+
+    # pylint: disable=unused-argument
+    def _get_enable_private_cluster(self, enable_validation: bool = False, **kwargs) -> bool:
+        """Internal function to obtain the value of enable_private_cluster.
+
+        This function supports the option of enable_validation. When enabled, if enable_private_cluster is specified
+        and load_balancer_sku is basic or api_server_authorized_ip_ranges is assigned, raise a
+        MutuallyExclusiveArgumentError; Otherwise when enable_private_cluster is not specified and disable_public_fqdn
+        or private_dns_zone is assigned, raise an InvalidArgumentValueError.
+
+        :return: bool
+        """
+        # read the original value passed by the command
+        enable_private_cluster = self.raw_param.get("enable_private_cluster")
+        # try to read the property value corresponding to the parameter from the `mc` object
+        if (
+            self.mc and
+            self.mc.api_server_access_profile and
+            self.mc.api_server_access_profile.enable_private_cluster is not None
+        ):
+            enable_private_cluster = self.mc.api_server_access_profile.enable_private_cluster
+
+        # this parameter does not need dynamic completion
+
+        # validation
+        if enable_validation:
+            if enable_private_cluster:
+                if self._get_load_balancer_sku(enable_validation=False) == "basic":
```

Comment:
ignore case

Reply:
Updated in commit [#62fd075](https://github.com/Azure/azure-cli/pull/19539/commits/62fd075d4d50739e6b1e47fa81de0ed79dcdf962).